### PR TITLE
Discord RPC Integration

### PR DESCRIPTION
**Pull request by Semphriss**
Replaces #1480 - This version should be much cleaner in terms of code.

Disabled by default, customisable, now you can show on your Discord profile your activity on SuperTux!

### CODE REVIEW

**Review by Zwatotem — CHANGES_REQUESTED** [review]
Please change the log level to info.

**Comment by Zwatotem** [comment]
Is the feedback that someone gave on Discord about levelset vs worldmap distinction addressed here?

**Comment by Semphriss** [comment]
That person was probably me (I don't remember if someone else complained about something else, but I remember I tripped on this thing and I complained on Discord about it).

Yes, normally this should be fine now.

File: CMakeLists.txt
Comment (Semphriss) [reply CMakeLists.txt:172]: I'll look into it, but there are CMake settings that rely on DISABLE_DISCORD. I'm not sure if I can merge the two, though

Comment (Semphriss) [reply CMakeLists.txt:172]: I can't make it that way because the way the system works, ENABLE_DISCORD must be unset, not false or off.

**Review by Zwatotem — APPROVED** [review]

File: CMakeLists.txt
Comment (Mathnerd314) [reply CMakeLists.txt:172]: It worked fine when I changed it: https://github.com/SuperTux/supertux/commit/b14abb07f8f0b6d1084b20e4b39782732d694eaa

Comment (Semphriss) [reply CMakeLists.txt:172]: It seems I can't compile it anymore with that commit; the Discord CMakeLists.txt no longer downloads RapidJson, for some reason. Not sure if it's related to your changes, but my last commit seems to work fine.

Also, I'd really suggest we keep the message saying whether or not Discord is being compiled, as it represents a proprietary/privacy concern (some pro-free software or pro-privacy distributors or users might want to disable it).

I'll look into this tomorrow and see if I can reproduce your changes, though. I'll also clean those unnecessary tabs on empty lines in discord.cpp :)

Comment (Semphriss) [reply CMakeLists.txt:172]: I owe you apologies; it appears I was wrong not only once, but twice! RapidJson not being found was due to my CMakeCache, and a CMake option set to Off indeed converts to no defined compiler variable in C++.

Now I'm just fixing some last compiler errors, and I'll push changes. (Do you know how I can silence a warning for an empty array?)

Comment (Semphriss) [reply CMakeLists.txt]: Fixed in latest commit; I'll wait for your approval before proceeding :)

**Pull request merged**
